pkg/util/xorm/sequence.go

```diff
@@ -0,0 +1,69 @@
+package xorm
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+)
+
+type sequenceGenerator struct {
+	db             *sql.DB
+	sequencesTable string
+}
+
+func newSequenceGenerator(db *sql.DB) *sequenceGenerator {
+	return &sequenceGenerator{
+		db:             db,
+		sequencesTable: "autoincrement_sequences",
+	}
+}
+
+func (sg *sequenceGenerator) Next(ctx context.Context, table, column string) (int64, error) {
+	// Current implementation fetches new value for each Next call.
+	key := fmt.Sprintf("%s:%s", table, column)
+
+	tx, err := sg.db.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSerializable})
+	if err != nil {
+		return 0, err
+	}
+
+	// TODO: "FOR UPDATE". (Somehow this doesn't seem to be supported in Spanner emulator?)
```

Comment:
To be fixed soon! https://github.com/GoogleCloudPlatform/cloud-spanner-emulator/issues/218#issuecomment-2705119157